graylog2-server/src/main/java/org/graylog2/database/PaginatedDbService.java

```diff
@@ -153,25 +152,18 @@ public int delete(String id) {
                                                                      DBSort.SortBuilder sort,
                                                                      int page,
                                                                      int perPage) {
-        final AtomicInteger total = new AtomicInteger(0);
+        final long total = streamQueryWithSort(query, sort).filter(filter).count();
 
-        try (final Stream<DTO> cursor = streamQueryWithSort(query, sort)) {
-            // First created a filtered stream
-            final Stream<DTO> dtoStream = cursor
-                    .filter(filter)
-                    .peek(dto -> total.incrementAndGet());
-
-            // Then use that filtered stream and only collect the entries according to page and perPage
-            Stream<DTO> resultStream = Stream.of(dtoStream).flatMap(stream -> stream);
-            if (perPage > 0) {
-                resultStream = resultStream.skip(perPage * Math.max(0, page - 1)).limit(perPage);
-            }
+        // Then use that filtered stream and only collect the entries according to page and perPage
+        Stream<DTO> resultStream = streamQueryWithSort(query, sort).filter(filter);
```

Comment:
Same as above, please make sure to close the stream.

Reply:
✅